lib/backend/spotifyService.dart

```diff
@@ -8,98 +8,83 @@ String client_secret = '2d710351a8ef47c18e29c875da325b7f';
 
 Client client = Client();
 
-Future<Response> initialAuth(String code) {
+Future<Response> initialAuth(String code, SharedPreferences prefs) {
   return client.post('https://accounts.spotify.com/api/token', headers:  clientHeaders(), body: preAuthBody(code)).then((Response response){
     final Map<String, dynamic> map = json.decode(response.body);
-    updateAuthToken(map['access_token']);
-    updateRefreshToken(map['refresh_token']);
-    updateExpireTime(map['expires_in'].toString());
-    updateCurrentUser(map['access_token']);
+    updateAuthToken(map['access_token'], prefs);
+    updateRefreshToken(map['refresh_token'], prefs);
+    updateExpireTime(map['expires_in'].toString(), prefs);
+    updateCurrentUser(map['access_token'], prefs);
   });
 }
 
-Future<Response> refreshAuth(String refreshToken) {
-  return client.post('https://accounts.spotify.com/api/token', headers: clientHeaders(), body: refreshBody(refreshToken)).then((Response result) {
-    final Map<String, dynamic> map = json.decode(result.body);
-    updateAuthToken(map['access_token']);
-    updateExpireTime(map['expires_in'].toString());
-    if (map.containsKey('refresh_token')) {
-      updateRefreshToken(map['refresh_token']);
-    }
-  });
+Future<Response> refreshAuth(String refreshToken, SharedPreferences prefs) {
+  return client.post('https://accounts.spotify.com/api/token', headers: clientHeaders(), body: refreshBody(refreshToken))
+      .then((Response result) {
+        final Map<String, dynamic> map = json.decode(result.body);
+        updateAuthToken(map['access_token'], prefs);
+        updateExpireTime(map['expires_in'].toString(), prefs);
+        if (map.containsKey('refresh_token')) {
+          updateRefreshToken(map['refresh_token'], prefs);
+        }
+    });
 }
 
-Future<String> getNowPlaying(String authToken) async {
-  String nowPlaying = 'Unable to get Now Playing';
+Future<Map<String, dynamic>> getNowPlaying(String authToken) async {
+  Map<String, dynamic> map;
   await client.get('https://api.spotify.com/v1/me/player/currently-playing', headers: authHeaders(authToken))
-      .then((Response response) {
-        nowPlaying = json.decode(response.body)['item']['name'];
-      }
-  );
-  return nowPlaying;
+      .then<Response>((Response response) => map = json.decode(response.body)['item']);
+  return map;
 }
 
-Future<String> getCurrentUser() async {
-  final SharedPreferences prefs = await SharedPreferences.getInstance();
-  return prefs.getString('current_user');
+Future<Map<String, dynamic>> getUserData(String authToken) async {
```

Comment:
Same with user id or whatever else you want in the object
https://developer.spotify.com/documentation/web-api/reference/users-profile/get-current-users-profile/